File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    clothes_dict = {}
    
    for name, type in clothes:
        if type in clothes_dict:
            clothes_dict[type] += 1
        else:
            clothes_dict[type] = 1
    
    result = 1
    
    for key in clothes_dict:
        result *= clothes_dict[key] + 1
    
    return result - 1

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    const count = new Map()
    
    for (const [_, type] of clothes) {
        count.set(type, (count.get(type) ?? 0) + 1)
    }
    
    let result = 1
    
    for (const value of count.values()) {
        result *= value + 1
    }

    return result - 1
}